vscode/src/azure/workspaceActions.ts

```diff
@@ -0,0 +1,374 @@
+
+  // Quick-pick if more than one
+  let workspace = workspaces.value[0];
+  if (workspaces.value.length > 1) {
+    const pickItems = workspaces.value.map((worksp) => ({
+      label: worksp.name,
+      detail: worksp.id,
+      selection: worksp,
+    }));
+    const choice = await vscode.window.showQuickPick(pickItems, {
+      title: "Select a workspace",
+    });
+    if (!choice) return;
+    workspace = choice.selection;
+  }
+
+  // Need to remove the first part of the endpoint
+  const fixedEndpoint =
+    workspace.properties.endpointUri?.replace(
+      `https://${workspace.name}.`,
+      "https://"
+    ) || "";
+
+  const result: WorkspaceConnection = {
+    id: workspace.id,
+    name: workspace.name,
+    endpointUri: fixedEndpoint,
+    tenantId,
+    connection: "AAD",
+    storageAccount: workspace.properties.storageAccount,
+    providers: workspace.properties.providers.map((provider) => ({
+      providerId: provider.providerId,
+      currentAvailability:
+        provider.provisioningState === "Succeeded"
+          ? "Available"
+          : "Unavailable",
+      targets: [], // Will be populated by a later query
+    })),
+    jobs: [],
+  };
+  if (log.getLogLevel() >= 5) {
+    log.trace(`Workspace object: ${JSON.stringify(result, null, 2)}`);
+  }
+
+  return result;
+}
+
+export async function getTokenForWorkspace(workspace: WorkspaceConnection) {
+  const workspaceAuth = await vscode.authentication.getSession(
+    "microsoft",
+    [scopes.quantum, `VSCODE_TENANT:${workspace.tenantId}`],
+    { createIfNone: true }
+  );
+  log.trace(`Got workspace token: ${JSON.stringify(workspaceAuth, null, 2)}`);
+  return workspaceAuth.accessToken;
+}
+
+// Reference for existing queries in Python SDK and Azure schema:
+// - https://github.com/microsoft/qdk-python/blob/main/azure-quantum/azure/quantum/_client/aio/operations/_operations.py
+// - https://github.com/Azure/azure-rest-api-specs/blob/main/specification/quantum/data-plane/Microsoft.Quantum/preview/2022-09-12-preview/quantum.json
+export async function queryWorkspace(workspace: WorkspaceConnection) {
+  const token = await getTokenForWorkspace(workspace);
+
+  const quantumUris = new QuantumUris(workspace.endpointUri, workspace.id);
+
+  const providerStatus: ResponseTypes.ProviderStatusList = await azureRequest(
+    quantumUris.providerStatus(),
+    token
+  );
+  if (log.getLogLevel() >= 5) {
+    log.trace(
+      `Got provider status: ${JSON.stringify(providerStatus, null, 2)}`
+    );
+  }
+
+  const supportedTargets = [
+    "quantinuum.sim.h1-1sc",
+    "quantinuum.sim.h1-1e",
+    "quantinuum.qpu.h1-1",
+    "quantinuum.sim.h1-2sc",
+    "quantinuum.sim.h1-2e",
+    "quantinuum.qpu.h1-2",
+    "quantinuum.sim.h2-1sc",
+    "quantinuum.sim.h2-1e",
+    "quantinuum.qpu.h2-1",
+    "rigetti.sim.qvm",
+    "rigetti.qpu.aspen-m-3",
+    "ionq.qpu-preview",
+    "ionq.qpu.aria-1-preview",
+    "ionq.qpu.aria-2-preview",
+    "ionq.simulator-preview",
+  ];
+
```

Comment:
We shouldn't be hard-coding any target ids like this. I thought there was a way we could filter to targets that accept the input type `qir.v1`?

Reply:
I don't see anywhere in the targets properties where it lists the input formats supported. (https://github.com/Azure/azure-rest-api-specs/blob/main/specification/quantum/data-plane/Microsoft.Quantum/preview/2022-09-12-preview/examples/providerStatus.json), and the only place inputDataFormat is used is in the job submission (https://github.com/Azure/azure-rest-api-specs/blob/main/specification/quantum/data-plane/Microsoft.Quantum/preview/2022-09-12-preview/quantum.json#L819).

We should request a service change so we can get this info back, and add a work item to make this dynamic once available. But in the meantime, I'd rather not allow jobs to be submitted to targets that we know won't work.

Reply:
> But in the meantime, I'd rather not allow jobs to be submitted to targets that we know won't work.

I understand the sentiment, but that's not consistent with how our service interactions work anywhere else. The user can pick any target and try to submit any content to it, it may fail (and fail quickly) if that input type is not compatible with that target. Almost of all of our other APIs work this way. Even though we can publish quickly, keeping a whitelist of targets in the extension itself means we have to update the extension (and the user has to install it) any time the allowed targets changes. I'd much rather leave the door open to users submitting to targets that reject the input than maintain this list here, even for an initial preview.

Reply:
Again, we'll remove once the service can tell us. But fow now, it's not like the Azure integration code is generic, it only submits to Azure as written, and we know what targets support QIR. I'd rather hardcode for now and give the same desired end-user experience. It seems counter-intuitive to let the user try and eventually get some error back if we know in advance which targets will and won't work and can prevent that up front.